.github/workflows/test-go.yml

```diff
@@ -55,10 +59,9 @@ jobs:
     permissions:
       id-token: write  # Note: this permission is explicitly required for Vault auth
       contents: read
-    name: ${{ inputs.packages }}
+    name: "${{ matrix.runner-index }}${{ inputs.extra-flags }}"
```

Comment:
Yeah, that's a good point. I only had it here as a placeholder, because I really couldn't figure out anything better/more meaningful. For now I settled on just passing a `name` parameter to the `test-go` workflow. With that in place, we will at least get to specify the name in the calling workflow and get something back that makes sense. I am quite keen to keep the 'test bucket' indexes in the name because that being a constant makes it easier to compare tests between runs, etc. That being said, I am totally open for any suggestions on how to deal better with these names.